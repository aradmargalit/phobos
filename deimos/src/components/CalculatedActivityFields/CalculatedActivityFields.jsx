import React from 'react';
import { Statistic } from 'antd';
import moment from 'moment';

const formatPace = (fraction) => moment().startOf('day').add(fraction, 'minutes').format('m:ss');
const singularize = (input) => input.slice(0, -1);

export default function CalculatedActivityFields({ activity }) {
  const { duration, distance, unit } = activity;

  const u = unit || 'miles';

  return (

    <Statistic title={`min / ${singularize(u)}`} value={formatPace(duration / distance)} />
  );
}
