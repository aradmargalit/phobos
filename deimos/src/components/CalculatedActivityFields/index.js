import CalculatedActivityFields from './CalculatedActivityFields';

export default CalculatedActivityFields;
