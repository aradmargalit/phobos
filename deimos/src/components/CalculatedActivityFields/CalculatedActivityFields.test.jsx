import { shallow } from 'enzyme';
import React from 'react';

import CalculatedActivityFields from './CalculatedActivityFields';

const generateComponent = (props) => {
  const defaultProps = {
    activity: {
      duration: { total: 100 },
      distance: 5,
      unit: 'miles',
    },
  };
  const mergedProps = { ...defaultProps, ...props };

  return shallow(<CalculatedActivityFields {...mergedProps} />);
};

describe('<CalculatedActivityFields />', () => {
  it('renders correctly', () => {
    expect(generateComponent()).toMatchSnapshot();
  });

  it('correctly singularizes unit', () => {
    expect(generateComponent().prop('title')).toEqual('min / mile');
  });

  it('correctly calculates pace', () => {
    expect(generateComponent().prop('value')).toEqual(20);
  });
});
