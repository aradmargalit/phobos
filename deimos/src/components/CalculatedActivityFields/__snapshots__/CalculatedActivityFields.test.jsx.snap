// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatedActivityFields /> renders correctly 1`] = `
<withConfigConsumer(Statistic)
  className="form-statistic"
  formatter={[Function]}
  title="min / mile"
  value={20}
/>
`;
